src/std/make.ss

```diff
@@ -22,222 +40,366 @@
 
 ;; buildspec: [<build> ...]
 ;;  <build>:
-;;    <module> ; module path string without extension
+;;   <module> ; module path string without extension
 ;;   (gxc: <module> gsc-opt ...)
 ;;   (gsc: <module> gsc-opt ...)
-;;   (ssi: <module>)
+;;   (ssi: <module> [gsc: (gsc-opt ...)] [static-include: file])
+;; TODO: provide an object-oriented way to extend the spec language,
+;; with methods spec-dependencies, spec-inputs, spec-outputs, spec-build
+;; and a macro so you can define all the methods coherently in one form,
+;; rather than manually keeping the forms in synch.
+
+(def (spec-type spec)
+  (match spec
+    ((? string? _) gxc:)
+    ([(? keyword? type) . _] type)
+    (else (error "Bad buildspec" spec))))
+
+(def (spec-file spec settings)
+  (match spec
+    ((? string? modf) (source-path modf ".ss" settings))
+    ([gxc: modf . opts] (source-path modf ".ss" settings))
+    ([gsc: modf . opts] (source-path modf ".scm" settings))
+    ([ssi: modf . deps] (source-path modf ".ssi" settings))
+    ([exe: modf . opts] (source-path modf ".ss" settings))
+    ([static-exe: modf . opts] (source-path modf ".ss" settings))
+    ([static-include: file] (static-file-path file settings))
+    ([copy: file] file)
+    (else
+     (error "Bad buildspec" spec))))
+
+(def (spec-inputs spec settings)
+  [(spec-file spec settings) (spec-dependencies spec settings) ...])
+
+(def (spec-dependencies spec settings)
+  (match spec
+    ([gxc: _ . opts] (xxc-compile-dep opts))
+    ([gsc: _ . opts] (xxc-compile-dep opts))
```

Comment:
Is this really the same compile method?

Reply:
Same dependency extraction method. I don't like the `dep:` keyword though. Maybe we can refactor it away, especially since these days it means "extra inputs" and not "dependencies".